main.cpp

```diff
@@ -92,6 +92,146 @@ good to go!
+ 
+ 4) make them return the correct primitive type. e.g. if you're implementing the FloatType::add function, your implementation would start like this:
+        float FloatType::add( float lhs, float rhs )
+ 
+ 5) declare a few instances of each class in main() and call the member functions with the appropriate arguments.
+        FloatType ft;
+        auto result = ft.add(3.2f, 23.f );
+ 
+ 6) print out those results using std::cout:
+        std::cout "result of ft.add(): " << result << std::endl;
+ 
+ 7) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
+ 
+ Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
+ 
+Make a pull request after you make your first commit and pin the pull request link to our DM thread.
+
+send me a DM to check your pull request
+
+ Wait for my code review.
+ */
+#include<iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs );
+    float subtract(float  lhs, float rhs );
+    float multiply(float  lhs, float rhs );
+    float divide(float  lhs, float rhs );
+};
+
+float FloatType::add(float lhs, float rhs )
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs )
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs )
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs )
+{
+    if (rhs == 0.0f)
+    {
+        std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
+}
+struct DoubleType
+{
+    double add(double lhs, double rhs );
+    double subtract(double lhs, double rhs );
+    double multiply(double lhs, double rhs );
+    double divide(double lhs, double rhs );
+};
+
+double DoubleType::add(double lhs, double rhs )
+{
+    return lhs + rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs )
+{
+    return lhs - rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs )
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs )
+{
+    if (rhs == 0.00)
+    {
+        std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+    }
+    return lhs / rhs;
+}
+
+struct IntType
+{
+    int add(int lhs, int rhs );
+    int subtract(int lhs, int rhs );
+    int multiply(int lhs, int rhs );
+    double divide(int lhs, int rhs );
+};
+
+int IntType::add(int lhs, int rhs )
+{
+    return lhs + rhs;
+}
+
+int IntType::subtract(int lhs, int rhs )
+{
+    return lhs - rhs;
+}
+
+int IntType::multiply(int lhs, int rhs )
+{
+    return lhs * rhs;
+}
+
+double IntType::divide(int lhs, int rhs )
+{
+    if (rhs != 0)
+    {
+        return lhs / rhs;
+    }
+    else
+    {
+        std::cout << "error, integer division by zero will crash the program! "<< std::endl;
+        std::cout<<"returning lhs" << std::endl;
+        return lhs;
+    }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise